railties/lib/rails/command/behavior.rb

```diff
@@ -0,0 +1,123 @@
+require "active_support"
+
+module Rails
+  module Command
+    module Behavior #:nodoc:
+      extend ActiveSupport::Concern
+
+      class_methods do
+        # Remove the color from output.
+        def no_color!
+          Thor::Base.shell = Thor::Shell::Basic
+        end
+
+        # Track all command subclasses.
+        def subclasses
+          @subclasses ||= []
+        end
+
+        protected
+
+          # This code is based directly on the Text gem implementation.
+          # Copyright (c) 2006-2013 Paul Battley, Michael Neumann, Tim Fletcher.
+          #
+          # Returns a value representing the "cost" of transforming str1 into str2.
+          def levenshtein_distance(str1, str2)
```

Comment:
I'm not sure we want to make this a public API. Maybe as `edit_distance_to('other string')`, but when would you want the distance directly versus just sort an array by it?

E.g. this might be more suitable:

```ruby
%w( some words with a distance ).sort_by_edit_distances_to('this string')
```